### ISSUE TITLE
[Feature] Show missing scenes in Performer / Studio page

### ISSUE THREAD

**Issue by Jabroni** [stashapp/stash#2264]
Was thinking after watching the amount of data that stashbox can have access too, it would be nice if we can see which scenes of an artists or studio we are missing (that the video is not locally stored). 

Maybe in a near feature we could do some basic nzb search for those missing scenes, and send them to download to a nzbget/sabnzb instance.  (I think this feature request  would be the first step to get into that direction)

**Comment by ALonelyJuicebox** [comment]
I feel strongly that Stash should in no way be associated with downloading content. Definitely disagree with the latter vision.

**Comment by coscosname** [comment]
That's true Stash should in no way be associated with downloading content but the option to add the missing scenes yourself would be useful.

This could work by adding a placeholder. Later when you add a scene to Stash there would be an option to merge the scene with the placeholder.

**Comment by Jabroni** [comment]
Technically speaking stash wouldn't have any illegal content. I was thinking in something like radarr/sonarr where stash would be just like an indexer that would know what it has and what its missing according to the artist/studio page, then lets say you want X missing scene, click it, it would do a search on nzb indexers you specified (with our account), and if found, send that nzb link to nzbget/sabnzbd to download and save to Y path where Stash would be able to scan it and add the scene to the DB.

**Comment by merqwury** [comment]
Figure out what's missing for a specific performer would be god-sent.

**Comment by SpedNSFW** [comment]
This is something I was actually thinking about today (not the downloading part, that's way out of Stash's scope and should _never_ be added).

Here are my thoughts (either so I don't forget if I want to implement this, or to the benefit of anyone else who does):
 - For performers with a Stash ID, have a new tab after "Scenes" called "Missing Scenes" or w/e
 - Within that tab, below all the filtering and page stuff, there should be a "Source" option, identical to the Tagger view, to select a Stash-box source.
 - From that source, get an array of all the scene Stash IDs attached to the performer, then remove any that are already present.
 - Display this page the same as any other Scene page, however clicking on a scene will redirect to the StashDB source.

**Comment by d3f113** [comment]
Not perfect, but there is a tampermonkey userscript for it, which you can use for stashdb. It would be better if that functionality would be integrated, but worth a try

**Comment by Gykes** [comment]
There are several plugins that can handle this as well as Whisparr.

**Comment by jjf3** [comment]
> This is something I was actually thinking about today (not the downloading part, that's way out of Stash's scope and should _never_ be added).
> 
> Here are my thoughts (either so I don't forget if I want to implement this, or to the benefit of anyone else who does):
> 
> * For performers with a Stash ID, have a new tab after "Scenes" called "Missing Scenes" or w/e
> * Within that tab, below all the filtering and page stuff, there should be a "Source" option, identical to the Tagger view, to select a Stash-box source.
> * From that source, get an array of all the scene Stash IDs attached to the performer, then remove any that are already present.
> * Display this page the same as any other Scene page, however clicking on a scene will redirect to the StashDB source.

Hey, stumbled on this issue while building something that tackles exactly this. I've got a working standalone tool that does the performer gap analysis externally rather than inside Stash — figured it's worth sharing since it's been an idea for a few years.
What it does so far:

Searches your local Stash library for a performer by name and pulls their StashDB ID from existing stash_ids
Scrapes the tracker's full tag-filtered search (paginated, gets all results not just the RSS cap) and cross-references against your local library using fuzzy token matching
Shows what you have (green), what's available to grab (found), and what's missing
One-click or batch send to qBittorrent with category and save path

The gap-finding logic is similar to what you're describing — pull what exists on the source, diff against what's local, display the delta. Main difference is it lives outside Stash as a Python-based local web tool rather than a Stash tab, which means the download integration isn't Stash's problem.
Working toward a Docker single-pane version with configurable tracker backends and Stash/Jellyfin/Plex library support. Early but functional — happy to share the repo if anyone wants to poke at it or contribute.
